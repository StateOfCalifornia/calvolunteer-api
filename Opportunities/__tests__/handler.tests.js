const handler = require('../handler');
const filters = require('../filters');

// ////////////////////////////////////////
//  validateRequest
// ////////////////////////////////////////
test('validateRequest should not return error message if location is set', () => {
    var req = {
        query: {
            location: 'a location'
        }
    }
    var result = handler.validateRequest(req);
    expect(result).toBeFalsy();
});

test('validateRequest should not return error message if virtual is set', () => {
    var req = {
        query: {
            virtual: true
        }
    }
    var result = handler.validateRequest(req);
    expect(result).toBeFalsy();
});

test('validateRequest should  return error message if location and virtual are not set', () => {
    var req = {
        query: {
        }
    }
    var result = handler.validateRequest(req);
    expect(result).toBeTruthy();
});

// ////////////////////////////////////////
//  getParameters
// ////////////////////////////////////////
test('getParameters should return proper json with true literal', () => {
    var req = {
        query: {
            ids: '1,2,3',
            location: 'california',
            pageNumber: 44,
            isCovid19: 'true',
            radius: 55,
            virtual: 'true',
            categories: '1,2,3',
            greatFor: 'kids,teens',
            skills: 'reading,writing',
            numberOfResults: 12,
            keywords: 'hospice care'
        }
    }
    var result = handler.getParameters(req);
    expect(result).toMatchSnapshot();
});

test('getParameters should return proper json with false literal', () => {
    var req = {
        query: {
            ids: '1,2,3',
            location: 'california',
            pageNumber: 22,
            isCovid19: 'false',
            radius: 12,
            virtual: 'false',
            categories: '1,2,3',
            greatFor: 'kids,teens',
            skills: 'reading,writing',
            numberOfResults: 14,
            keywords: 'tutoring'
        }
    }
    var result = handler.getParameters(req);
    expect(result).toMatchSnapshot();
});

test('getParameters should set defaults', () => {
    var req = {
        query: {
            location: 'california',
        }
    }
    var result = handler.getParameters(req);
    expect(result).toMatchSnapshot();
});
test('getParameters should set defaults with virtual', () => {
    var req = {
        query: {
            virtual: 'true',
        }
    }
    var result = handler.getParameters(req);
    expect(result).toMatchSnapshot();
});
// ////////////////////////////////////////
//  processOpportunities
// ////////////////////////////////////////

jest.mock('../filters');
test('processOpportunities should call filter.CustomRequiredFields', () => {
    var pre = [{},{},{}];
    handler.processOpportunities(pre);
    expect(filters.CustomRequiredFields).toHaveBeenCalledTimes(1);
});

test('processOpportunities should not call filter.CustomRequiredFields if skipCustom is truthy', () => {
    var pre = [{},{},{}];
    var skipCustom = 1;
    handler.processOpportunities(pre, skipCustom);
    expect(filters.CustomRequiredFields).toHaveBeenCalledTimes(0);
});
test('processOpportunities should call filter.CAOrg', () => {
    var pre = [{},{},{}];
    handler.processOpportunities(pre);
    expect(filters.CAOrg).toHaveBeenCalledTimes(1);
});
