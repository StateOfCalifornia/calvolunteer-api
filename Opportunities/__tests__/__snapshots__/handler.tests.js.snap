// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getParameters should return proper json with false literal 1`] = `
Object {
  "categories": "1,2,3",
  "covid19Flag": "",
  "greatFor": "kids,teens",
  "ids": "1,2,3",
  "key": "",
  "keywords": "tutoring",
  "location": "california",
  "numberOfResults": 14,
  "pageNumber": 22,
  "radius": 12,
  "skills": "\\"reading\\",\\"writing\\"",
  "virtual": false,
}
`;

exports[`getParameters should return proper json with true literal 1`] = `
Object {
  "categories": "1,2,3",
  "covid19Flag": "covid19",
  "greatFor": "kids,teens",
  "ids": "1,2,3",
  "key": "",
  "keywords": "hospice care",
  "location": "california",
  "numberOfResults": 12,
  "pageNumber": 44,
  "radius": 55,
  "skills": "\\"reading\\",\\"writing\\"",
  "virtual": true,
}
`;

exports[`getParameters should set defaults 1`] = `
Object {
  "categories": "",
  "covid19Flag": "",
  "greatFor": "",
  "ids": "",
  "key": "",
  "keywords": "",
  "location": "california",
  "numberOfResults": 100,
  "pageNumber": 1,
  "radius": 20,
  "skills": "",
  "virtual": false,
}
`;

exports[`getParameters should set defaults with virtual 1`] = `
Object {
  "categories": "",
  "covid19Flag": "",
  "greatFor": "",
  "ids": "",
  "key": "",
  "keywords": "",
  "location": "",
  "numberOfResults": 100,
  "pageNumber": 1,
  "radius": 20,
  "skills": "",
  "virtual": true,
}
`;
